scenes/e2e/screenshot.ts

```diff
@@ -0,0 +1,44 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import {by, element} from 'protractor';
+
+const OUTPUT_DIR = path.join(__dirname, 'screenshots');
+
+
+export class Screenshot {
+  id: string;
+
+  /** The filename used to store the screenshot. */
+  get filename(): string {
+    return this.id
+        .toLowerCase()
+        .replace(/\s/g, '_')
+        .replace(/[^/a-z0-9_]+/g, '')
+      + '.scene.png';
+  }
+
+  /** The full path to the screenshot */
+  get fullPath(): string {
+    return path.resolve(OUTPUT_DIR, this.filename);
+  }
+
+  constructor(id: string) {
+    this.id = id;
+    element(by.id('wrapper')).takeScreenshot().then(png => this.storeScreenshot(png));
+  }
+
+  /** Replaces the existing screenshot with the newly generated one. */
+  storeScreenshot(png: any) {
+    if (!fs.existsSync(OUTPUT_DIR)) {
+      fs.mkdirSync(OUTPUT_DIR, '744');
```

Comment:
does this image really need execute permissions?
```suggestion
      fs.mkdirSync(OUTPUT_DIR, '444');
```